ndbench-es-plugins/src/main/java/com/netflix/ndbench/plugin/es/EsAutoTuner.java

```diff
@@ -0,0 +1,78 @@
+package com.netflix.ndbench.plugin.es;
+
+import com.netflix.ndbench.api.plugin.NdBenchMonitor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Date;
+
+
+class EsAutoTuner {
+    private final ConstantStepWiseRateIncreaser rateIncreaser;
+    private final Float autoTuneFailureRatioThreshold;
+
+
+    private static final Logger logger = LoggerFactory.getLogger(EsAutoTuner.class);
+
+    /**
+     * // captures  time of first auto-tune recommendation request (made via a call to recommendNewRate);
+     * TODO - update comment
+     * Ignore the  possible race condition that arises if multiple threads call this method at around the same time
+     * when an instance is an uninitialized state. Two (or maybe more) instance of ConstantStepWiseRateIncreaser
+     * may be concurrently built, but the system clock time to rate mappings will be close enough that it
+     * won't really matter -- auto-tuning behavior will work as expected.
+     */
+    private volatile long timeOfFirstAutoTuneRequest = -1;
+
+    private volatile boolean writeFailureThresholdReached = false;
```

Comment:
Use AtomicBoolean instead?

Reply:
Thank you for pointing this out. I tried educating myself a little more on atomicboolean vs. volatile.
(e.g., here: https://stackoverflow.com/questions/3786825/volatile-boolean-vs-atomicboolean )

I came away w/ the conclusion that there are some cases where AtomicBoolean  is better than 
volatile, but that in my use of this flag it doesn't make much difference. The reason is
I am not...

    trying to invert the boolean variable (flip from t->f or f->t) 
    (if i were, then  AtomicBoolean would be the way to go.. )

    and i am not doing the following either (see stack overflow answer):

        >>>
        You can't do compareAndSet, getAndSet as atomic operation with volatile boolean (unless of course you synchronize it).
        >>>


the way I am using it is simply as a flag that is initially off (threshold not reached) and that can be 
set to on idempotently by any number of threads...  So for this useage i think volatile is OK.
I don't think there is a difference. If you disagree I'd like to understand why, 'cause I will probably learn
something. 

And at the end of the day, even if I still don't see the value of changing it I'll change it anyway
if you really want me to, as I think AtomicBoolean would be fine here as well.